File: 2762. Continuous Subarrays/solution.py
class Solution:
    def continuousSubarrays(self, nums):
        n = len(nums)
        left = 0
        range_min = float('inf')
        range_max = float('-inf')
        count = 0
        right = 0
        while right < n:
            range_min = min(range_min,nums[right])
            range_max = max(range_max,nums[right])
            if range_max - range_min > 2:
                wind_size = right - left
                count += ((wind_size)*(wind_size + 1))//2
                left = right
                range_min = nums[right]
                range_max = nums[right]
                while left > 0 and abs(nums[right] - nums[left-1]) <= 2:
                    left -= 1
                    range_min = min(range_min,nums[left])
                    range_max = max(range_max,nums[left])
                if left < right: 
                    new_wind = right - left     
                    count -= ((new_wind)*(new_wind+1))//2
            right += 1
        count += ((right - left)*(right - left + 1))//2
        return count            

File: 2762. Continuous Subarrays/solution.rs
impl Solution {
    pub fn continuous_subarrays(nums: Vec<i32>) -> i64 {
        let n = nums.len();
        let mut left = 0;
        let mut right = 0;
        let mut count = 0;
        let mut min_range = i64::MAX;
        let mut max_range = i64::MIN;
        while right < n{
            max_range = std::cmp::max(max_range,nums[right] as i64);
            min_range = std::cmp::min(min_range,nums[right] as i64);
            if max_range - min_range > 2{
                let wind_size = (right - left) as i64;
                count += ((wind_size)*(wind_size+1))/2;
                min_range = nums[right] as i64;
                max_range = nums[right] as i64;
                left = right;
                while left > 0 && (nums[right]-nums[left-1]).abs() <= 2{
                    left -= 1;
                    max_range = std::cmp::max(max_range,nums[left] as i64);
                    min_range = std::cmp::min(min_range,nums[left] as i64);
                    
                }
                if left < right{
                    let win_size = (right - left) as i64;
                    count -= ((win_size)*(win_size+1))/2;
                }

            } right += 1;

        }
        let wind_size = (right - left) as i64;
        return count + ((wind_size)*(wind_size+1))/2
        
    }
}